test/workers/delete_object_hierarchy_worker_test.rb

```diff
@@ -40,24 +37,36 @@ def hierarchy_worker
     DeleteObjectHierarchyWorker
   end
 
-  class ObjectDoesNotExistAnymoreTest < DeleteObjectHierarchyWorkerTest
+  class DeleteObjectHierarchyWorkerWhenDoesNotHaveAssociationsTest < ActiveSupport::TestCase
+    def test_execute_success_when_empty_batch
+      object = FactoryBot.create(:metric)
+      worker = DeleteObjectHierarchyWorker.new
+      worker.instance_variable_set(:@object, object)
+      caller_worker_hierarchy = %w[HTestClass123 HTestClass1123]
+      worker.instance_variable_set(:@caller_worker_hierarchy, caller_worker_hierarchy)
+      worker.expects(:on_success).with(anything, {'object_global_id' => object.to_global_id, 'caller_worker_hierarchy' => caller_worker_hierarchy})
```

Comment:
This is failing because now it should be `worker.expects(:on_complete).with(anything, {'object_global_id' => object.to_global_id, 'caller_worker_hierarchy' => caller_worker_hierarchy})`
For this:
https://github.com/3scale/porta/pull/522/files#diff-ff63dfaf8bfa4a850361d41a988711a1R70